test/integration/tags/render_tag_test.rb

```diff
@@ -168,6 +168,28 @@ def test_includes_will_not_render_inside_nested_sibling_tags
     assert_template_result('include usage is not allowed in this contextinclude usage is not allowed in this context', '{% render "nested_render_with_sibling_include" %}')
   end
 
+  def test_render_tag_uses_correct_disabled_tags_instance
+    Liquid::Template.file_system = StubFileSystem.new(
+      'foo' => 'bar',
+      'test_include' => '{% include "foo" %}'
+    )
+
+    source = "{% render 'test_include' %}"
+    tag = Render.parse("render", source, Tokenizer.new("", true), ParseContext.new)
+
+    buffer = +""
+    context = Context.new
+    disabled_tags = DisabledTags.new
+    context.registers[:disabled_tags] = disabled_tags
+
+    context.registers[:disabled_tags].disable(tag.disabled_tags) do
+      tag.render_tag(context, buffer)
+      assert_equal('include usage is not allowed in this context', buffer)
+    end
+
+    assert_equal(disabled_tags, context.registers[:disabled_tags])
+  end
```

Comment:
As discussed IRL, I think this test should be moved to `TemplateTest` instead.

Also, we should avoid copying `context.registers[:disabled_tags].disable` which is internal logic from `BlockBody` and instead consider something like

```ruby
def test_render_tag_uses_correct_disabled_tags_instance
    Liquid::Template.file_system = StubFileSystem.new(
      'foo' => 'bar',
      'test_include' => '{% include "foo" %}'
    )

    disabled_tags = DisabledTags.new
    context = Context.build(registers: { disabled_tags: disabled_tags })

    source = "{% render 'test_include' %}"
    parse_context = Liquid::ParseContext.new(line_numbers: true, error_mode: :strict)
    document = Document.parse(Liquid::Tokenizer.new(source, true), parse_context)

    assert_equal("include usage is not allowed in this context", document.render(context))
  end
```

Reply:
Agreed! 👍